sdk/python/lib/pulumi/stack_reference.py

```diff
@@ -63,7 +63,7 @@ class StackReference(CustomResource):
     The name of the referenced stack.
     """
 
-    outputs: Output[dict]
+    outputs: Output[dict[str, Any]]
```

Comment:
This needs to be Dict, but I'd suggest just rebasing on https://github.com/pulumi/pulumi/pull/16956 and then we'll merge this for tests.